src/cargo/core/shell.rs

```diff
@@ -147,27 +147,66 @@ impl MultiShell {
 }
 
 impl Shell {
-    pub fn create(out: Box<Write + Send>, config: ShellConfig) -> Shell {
+    pub fn create<T: FnMut() -> Box<Write + Send>>(mut out_fn: T, config: ShellConfig) -> Shell {
+        let term = match Shell::get_term(out_fn()) {
+            Ok(t) => t,
+            Err(_) => NoColor(out_fn())
+        };
+
+        Shell {
+            terminal: term,
+            config: config,
+        }
+    }
+
+    /// Create a shell in such a manner that it cannot possibly fail
+    /// Use this after `create` has failed.
+    pub fn create_fallback(out: Box<Write + Send>, config: ShellConfig) -> Shell {
```

Comment:
Yea it seems I forgot to append that change to my commit. I'll add these changes!